generic_config_updater/generic_updater.py

```diff
@@ -0,0 +1,469 @@
+
+        sy._cropConfigDB()
+
+        return sy.jIn
+
+    def __create_and_connect_config_db(self):
+        if self.default_config_db_connector != None:
+            return self.default_config_db_connector
+
+        db_kwargs = dict()
+        config_db = ConfigDBConnector(**db_kwargs)
+        config_db.connect()
+        return config_db
+
+class DryRunConfigWrapper(ConfigWrapper):
+    # TODO: implement DryRunConfigWrapper
+    # This class will simulate all read/write operations to ConfigDB on a virtual storage unit.
+    pass
+
+class PatchWrapper:
+    def __init__(self, config_wrapper = ConfigWrapper()):
+        self.config_wrapper = config_wrapper
+
+    def validate_config_db_patch(self, patch):
+        config_db = {}
+        for operation in patch:
+            tokens = operation['path'].split('/')[1:]
+            if len(tokens) == 0: # Modifying whole config_db
+                tables_dict = {table_name: {} for table_name in operation['value']}
+                config_db.update(tables_dict)
+            elif not tokens[0]: # Not empty
+                raise Exception("Table name in patch cannot be empty")
+            else:
+                config_db[tokens[0]] = {}
+
+        cropped_config_db = self.config_wrapper.crop_tables_without_yang(config_db)
+
+        # valid if no tables dropped during cropping
+        return len(cropped_config_db.keys()) == len(config_db.keys())
+
+    def verify_same_json(self, expected, actual):
+        # patch will be [] if no diff, [] evaluates to False
+        return not jsonpatch.make_patch(expected, actual)
+
+    def generate_patch(self, current, target):
+        return jsonpatch.make_patch(current, target)
+
+    def simulate_patch(self, patch, jsonconfig):
+        return patch.apply(jsonconfig)
+
+    def convert_config_db_patch_to_sonic_yang_patch(self, patch):
+        if not(self.validate_config_db_patch(patch)):
+            raise Exception(f"Given patch is not valid")
+
+        current_config_db = self.config_wrapper.get_config_db_as_json()
+        target_config_db = self.simulate_patch(patch, current_config_db)
+
+        current_yang = self.config_wrapper.convert_config_db_to_sonic_yang(current_config_db)
+        target_yang = self.config_wrapper.convert_config_db_to_sonic_yang(target_config_db)
+
+        return self.generate_patch(current_yang, target_yang)
+
+class ConfigFormat(Enum):
+    SONICYANG = 1
+    CONFIGDB = 2
+
+class PatchApplier:
+    def __init__( \
+        self, \
+            patchorderer = PatchOrderer(), \
+                changeapplier = ChangeApplier(), \
+                    config_wrapper = ConfigWrapper(), \
+                        patch_wrapper = PatchWrapper()):
+        self.patchorderer = patchorderer
+        self.changeapplier = changeapplier
+        self.config_wrapper = config_wrapper
+        self.patch_wrapper = patch_wrapper
+
+    def apply(self, patch):
+        # 1. Get old config as SONiC Yang
```

Comment:
>1.  [](http://example.com/codeflow?start=10&length=3)

Normally we don't use numbered list in code comment. It's hard to maintain if you can code, add complex control logic. It is a subject opionion.